test/RelayHubGas.test.ts

```diff
@@ -0,0 +1,1471 @@
+            const url = 'http://relay.com';
+            const message = 'Enveloping RelayHub';
+            const messageWithNoParams = 'Method with no parameters';
+
+            let relayRequest: RelayRequest;
+            let encodedFunction: string;
+            let signatureWithPermissiveVerifier: string;
+
+            beforeEach(async function () {
+                await relayHubInstance.stakeForAddress(relayManager, 1000, {
+                    value: ether('2'),
+                    from: relayOwner
+                });
+
+                // truffle-contract doesn't let us create method data from the class, we need an actual instance
+                encodedFunction = recipientContract.contract.methods
+                    .emitMessage(message)
+                    .encodeABI();
+
+                await relayHubInstance.addRelayWorkers([relayWorker], {
+                    from: relayManager
+                });
+                await relayHubInstance.registerRelayServer(url, {
+                    from: relayManager
+                });
+                relayRequest = cloneRelayRequest(sharedRelayRequestData);
+                relayRequest.request.data = encodedFunction;
+
+                const dataToSign = new TypedRequestData(
+                    chainId,
+                    forwarder,
+                    relayRequest
+                );
+                signatureWithPermissiveVerifier = getLocalEip712Signature(
+                    dataToSign,
+                    gaslessAccount.privateKey
+                );
+            });
+
+            context('with funded verifier', function () {
+                let signature;
+
+                let misbehavingVerifier: TestVerifierConfigurableMisbehaviorInstance;
+
+                let signatureWithMisbehavingVerifier: string;
+                let relayRequestMisbehavingVerifier: RelayRequest;
+                const gas = 4e6;
+
+                beforeEach(async function () {
+                    misbehavingVerifier =
+                        await TestVerifierConfigurableMisbehavior.new();
+
+                    let dataToSign = new TypedRequestData(
+                        chainId,
+                        forwarder,
+                        relayRequest
+                    );
+
+                    signature = getLocalEip712Signature(
+                        dataToSign,
+                        gaslessAccount.privateKey
+                    );
+
+                    relayRequestMisbehavingVerifier =
+                        cloneRelayRequest(relayRequest);
+                    relayRequestMisbehavingVerifier.relayData.callVerifier =
+                        misbehavingVerifier.address;
+
+                    dataToSign = new TypedRequestData(
+                        chainId,
+                        forwarder,
+                        relayRequestMisbehavingVerifier
+                    );
+                    signatureWithMisbehavingVerifier = getLocalEip712Signature(
```

Comment:
signatureWithPermissiveVerifier -> variable is not being used

Reply:
fixed